### PR TITLE
fix: allow user to choose generation splits

### PR DESCRIPTION
**Pull request by elboy3**
https://app.shortcut.com/galileo/story/7844/dq-allow-choosing-on-which-split-to-generate?vc_group_by=day&ct_workflow=all&cf_workflow=500000005

### CODE REVIEW

File: dataquality/integrations/seq2seq/hf.py [hunk 1]
```diff
@@ -92,8 +92,12 @@ def watch(
     seq2seq_logger_config.model = model
     seq2seq_logger_config.generation_config = generation_config
 
-    generation_splits = {Split.validation, Split.test}
-    if generate_training_data:
-        generation_splits.add(Split.training)
+    generation_splits = {Split.test}
+    for split in generate_on:
+        if split not in Split.get_valid_keys():
+            warn(f"Ignoring invalid generation split {split}")
```
Comment (bogdan-galileo) [inline dataquality/integrations/seq2seq/hf.py:98]: What do you think about adding another sentence like: `The valid splits are ...` ?

**Review by bogdan-galileo — APPROVED** [review]
I know that my approval doesn't count to allow merging the PR, but eh 🤷

File: dataquality/integrations/seq2seq/hf.py [hunk 2]
```diff
@@ -92,8 +92,15 @@ def watch(
     seq2seq_logger_config.model = model
     seq2seq_logger_config.generation_config = generation_config
 
-    generation_splits = {Split.validation, Split.test}
-    if generate_training_data:
-        generation_splits.add(Split.training)
+    generation_splits = {Split.test}
+    for split in generate_on:
```
Comment (dcaustin33) [inline dataquality/integrations/seq2seq/hf.py:96]: not intuitive arg without docstring

**Review by dcaustin33 — APPROVED** [review]

File: dataquality/integrations/seq2seq/hf.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def watch(
     model: PreTrainedModel,
     tokenizer: PreTrainedTokenizerFast,
     generation_config: GenerationConfig,
-    generate_training_data: bool = False,
+    generatation_splits: List[str] = [],
```
Comment (setu4993) [inline dataquality/integrations/seq2seq/hf.py:69]: I'd recommend doing something like:
```suggestion
    generatation_splits: Optional[List[str]] = None,
    ...
    generatation_splits = generatation_splits or []
```

Mutable objects!

**Review by setu4993 — APPROVED** [review]
One request, rest looks good.

**Pull request merged**